x-pack/plugins/enterprise_search/public/applications/app_search/components/schema/views/schema.tsx

```diff
@@ -38,13 +44,24 @@ export const Schema: React.FC = () => {
           { defaultMessage: 'Add new fields or change the types of existing ones.' }
         )}
         rightSideItems={[
-          <EuiButton fill>
+          <EuiButton
+            fill
+            disabled={!hasSchemaChanged}
+            isLoading={isUpdating}
+            onClick={() => updateSchema()}
+            data-test-subj="updateSchemaButton"
+          >
             {i18n.translate(
               'xpack.enterpriseSearch.appSearch.engine.schema.updateSchemaButtonLabel',
               { defaultMessage: 'Update types' }
             )}
           </EuiButton>,
-          <EuiButton color="secondary">
+          <EuiButton
+            color="secondary"
+            disabled={isUpdating}
```

Comment:
I opted for `disabled` over `isLoading` here because it doesn't make a whole ton of sense that the open modal button gets a loading spinner whenever schema updates (rather than the user simply not being able to add a new field during updates).